### ISSUE TITLE
Cannot restore encrypted images

### ISSUE THREAD

**Issue by 9YbQiuEohUu1** [7LPdWcaW/GrowTracker-Android#140]
I turned encryption on, then uninstalled the app. After i reinstalled it, it displayed all my plant data (to my surprise, as I thought it would be in an encrypted state), but not the photos. I then encrypted it again with the same passphrase, after which previews showed up for some plants in the garden screen but none of the photos became visible (even those used for preview). Is it still possible to recover those photos?

**Comment by 7LPdWcaW** [comment]
ummmm, sort of. there's a logic error with reinstalling the app and re-enabling encryption which causes the images to be encrypted again (double encrypted). I wrote a little java application that can decrypt files within a folder, but I cant guarantee recovery. Encryption within the app needs a serious re-write as there are many issues and scaling issues with how it encrypts. I will post the java application soon when I can

**Comment by 9YbQiuEohUu1** [comment]
As a workaround, I've tried unencrypting everything back, moving images to a different directory, encrypting the data with the same passphrase, then closing the app, and moving the images back to DCIM/GrowTracker. Now some images do show up but most do not, and some of the displayed ones seem to be broken (display only half of the image) :(

**Comment by 7LPdWcaW** [comment]
Yeah, apologies. I myself have lost many photos because of this. I plan on re-writing the encryption mode to prevent this.

Here is a decrypter jar. To use, open terminal or command prompt and type `java -jar Decrypter.jar -p=<path to folder with encrypted files> -o=<path to decrypt to> --encrypt-key=<encryption key>`

make sure paths end in a `/` (or `\` for windows). If they are double encrypted, you'll need to do it twice (once on the original folder, then again on the first decrypted folder). If the passphrase is wrong, you'll get an error like `java.io.IOException: javax.crypto.IllegalBlockSizeException: Input length must ...`)

I hope you're able to recover them

[Decrypter.zip](https://github.com/7LPdWcaW/GrowTracker-Android/files/3643256/Decrypter.zip)

**Comment by 9YbQiuEohUu1** [comment]
Thank you, I'll give it a try.

**Comment by 9YbQiuEohUu1** [comment]
Okay, progress report. I was able to decrypt all of the images. Some of them turned out corrupted, showing only a part of the image. There were also two *.jpg.temp files which I renamed to *.jpg and successfully decrypted - luckily, they turned out undamaged duplicates of the corrupted files. I ended up with only two broken photos and thought that was okay. However, after I pasted them back into DCIM/GrowTracker, the app showed four corrupted photos. Interestingly, the other two files themselves are ok when opened using an external app, so it is GrowTracker that fails to display them correctly for some reason. I had the same issue with uploading new photos - sometimes they end up uploaded only partially. However, I'm not sure whether that's the app's fault as I run it on an antiquated tablet with a capricious custom rom. In any case, thanks a lot for all the help. Hope you'll find the feedback useful.
